script/setup

```diff
@@ -1,11 +1,26 @@
-#!/bin/sh
+#!/bin/bash
 # Setups the repository.
 
 # Stop on errors
 set -e
 
 cd "$(dirname "$0")/.."
+
+mkdir -p config
+
+python3 -m venv venv
+source venv/bin/activate
```

Comment:
Do we need to document that if the developer wants to maintain another environment the developer can't use this script?

Reply:
What do you mean with maintain another environment? Like have one virtual env that mixes both HA deps and deps of another project? That isn't supported. For installing external deps, people should do `pip3 install -e /path/to/project`

Reply:
No I mean use another venv tool. Eg I use virtualvenv and virtualenvwrapper to manage my venvs. So this script won't be compatible with that setup.

Reply:
I think that those users should look into the script and figure out it's just `pip3 install -e .`. `venv` is official part of Python and should be the preferred venv.

Reply:
I guess we can also add support for an ENV var to skip the virtual env.

Reply:
Can't we just check for $VIRTUAL_ENV being set, and abort if not. It should cover most bases.

Reply:
If that is a thing, yes.

Reply:
It is set by activate script. May not be set if a binary is run directly in venv, but that should be case we need to care about.